sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,485 @@
+		return [kiosk, cap];
+	}
+
+	/**
+	 * Single function way to create a kiosk, share it and transfer the cap to the specified address.
+	 */
+	createAndShare(txb: TransactionBlock, address: string) {
+		let cap = kioskTx.createKioskAndShare(txb);
+		txb.transferObjects([cap], txb.pure(address, 'address'));
+	}
+
+	/**
+	 * Should be called only after `create` is called.
+	 * It shares the kiosk & transfers the cap to the specified address.
+	 */
+	shareAndTransferCap(
+		txb: TransactionBlock,
+		kiosk: TransactionArgument,
+		cap: TransactionArgument,
+		address: string,
+	) {
+		kioskTx.shareKiosk(txb, kiosk);
+		txb.transferObjects([cap], txb.pure(address, 'address'));
+	}
+
+	/**
+	 * Wraps a kiosk transaction that depends on `kioskOwnerCap`.
+	 * @param txb The Transaction Block
+	 * @param ownerCap The `ownerCap` object as returned from the `getOwnedKiosk` function
+	 * @param callback The function you want to execute with the ownerCap.
+	 */
+	async ownedKioskTx(
+		txb: TransactionBlock,
+		callback: (kiosk: TransactionArgument, capObject: TransactionArgument) => Promise<void>,
+	): Promise<void> {
+		this.#verifyCapIsSet();
+		let [kiosk, capObject, returnPromise] = this.getOwnerCap(txb);
+
+		await callback(kiosk, capObject);
+
+		this.returnOwnerCap(txb, capObject, returnPromise);
+	}
+
+	/**
+	 * Get an addresses's owned kiosks.
+	 * @param address The address for which we want to retrieve the kiosks.
+	 * @returns An Object containing all the `kioskOwnerCap` objects as well as the kioskIds.
+	 */
+	async getOwnedKiosks(address: string): Promise<OwnedKiosks> {
+		let personalPackageId = personalKioskAddress[this.network];
+		return getOwnedKiosks(this.client, address, {
+			personalKioskType: personalPackageId
+				? `${personalPackageId}::personal_kiosk::PersonalKioskCap`
+				: '',
+		});
+	}
+
+	/**
+	 * Fetches the kiosk contents.
+	 * @param kioskId
+	 * @param options
+	 * @returns
+	 */
+	async getKiosk(kioskId: string, options: FetchKioskOptions): Promise<KioskData> {
+		return (
+			await fetchKiosk(
+				this.client,
+				kioskId,
+				{
+					limit: 1000,
+				},
+				options,
+			)
+		).data;
+	}
+
+	/**
+	 * Query the Transfer Policy(ies) for type `T`.
+	 * @param itemType The Type we're querying for (E.g `0xMyAddress::hero::Hero`)
+	 */
+	async getTransferPolicies(itemType: string) {
+		return queryTransferPolicy(this.client, itemType);
+	}
+
+	/**
+	 * A function to purchase and resolve a transfer policy.
+	 * If the transfer policy has the `lock` rule, the item is locked in the kiosk.
+	 * Otherwise, the item is placed in the kiosk.
+	 * @param txb The Transaction Block
+	 * @param item The item {type, objectId, price}
+	 * @param options Currently has `extraArgs`, which can be used for custom rule resolvers.
+	 */
+	async purchaseAndResolve(
+		txb: TransactionBlock,
+		itemType: string,
+		itemId: string,
+		price: string,
+		kiosk: ObjectArgument,
+		ownedKiosk: ObjectArgument,
+		ownedKioskCap: ObjectArgument,
```

Comment:
likewise, can one of them be derived from another?

Reply:
Not necessarily. We could be in a situation when we create a new kiosk, so these are of type `TransactionArgument`